lib/Echidna/Output/Corpus.hs

```diff
@@ -26,3 +31,35 @@ loadTxs dir = do
   putStrLn ("Loaded " ++ show (length txSeqs) ++ " transaction sequences from " ++ dir)
   pure txSeqs
   where readCall f = decodeStrict <$> BS.readFile f
+
+-- save to corpus in the background while tests are running
+-- returns a channel that gets sent () when the process is done
+runCorpusSaver :: Env -> IO (Chan ())
```

Comment:
can we use MVar instead of Chan here?

Reply:
Is there a way to wait for MVar to be written to without doing an infinite loop? I used a Chan because you can wait for it using `readChan` instead of getting locked in a `while (thead is not finished)` loop

Reply:
yeah, it should be possible